src/librustc_incremental/persist/load.rs

```diff
@@ -103,41 +139,66 @@ fn delete_dirty_work_product(tcx: TyCtxt,
     work_product::delete_workproduct_files(tcx.sess, &swp.work_product);
 }
 
-pub fn load_dep_graph(sess: &Session) -> PreviousDepGraph {
-    let empty = PreviousDepGraph::new(SerializedDepGraph::new());
+/// Launch a thread and load the dependency graph in the background.
+pub fn load_dep_graph(sess: &Session) -> std::thread::JoinHandle<LoadResult<PreviousDepGraph>> {
+    let is_incremental = sess.opts.incremental.is_some();
+    let report_incremental_info = sess.opts.debugging_opts.incremental_info;
+    let path = dep_graph_path_from(&sess.incr_comp_session_dir());
+    let expected_hash = sess.opts.dep_tracking_hash();
 
-    if sess.opts.incremental.is_none() {
-        return empty
-    }
+    let aux = move || {
```

Comment:
Could you make this just one closure that is directly passed to `thread::spawn()`? Having one closure nested inside of another feels a bit roundabout.